modules/32_power/IntC/equations.gms

```diff
@@ -278,9 +278,12 @@ q32_flexPriceBalance(t,regi)$(cm_FlexTaxFeedback eq 1)..
 *** while inflexible technologies are penalized (v32_flexPriceShare > 1).  
 *** Flexibility tax is switched only if cm_flex_tax = 1 and is active from 2025 onwards. 
 q32_flexAdj(t,regi,te)$(teFlexTax(te))..
-	vm_flexAdj(t,regi,te) 
-	=e=
-	(1-v32_flexPriceShare(t,regi,te)) * pm_SEPrice(t,regi,"seel")$((cm_flex_tax eq 1) AND (t.val ge 2025) and (pm_SEPrice(t,regi,"seel") gt 0))
+  vm_flexAdj(t,regi,te) 
+  =e=
+  (1-v32_flexPriceShare(t,regi,te)) 
+  * ( pm_SEPrice(t,regi,"seel")$((cm_flex_tax eq 1) AND (t.val ge 2025) and (pm_SEPrice(t,regi,"seel") gt 0) and (pm_SEPrice(t,regi,"seel") lt 2))
+      + 2$((cm_flex_tax eq 1) AND (t.val ge 2025) and (pm_SEPrice(t,regi,"seel") gt 0) and (pm_SEPrice(t,regi,"seel") ge 2)) !! limit maximum applied electricity price to 230$/MWh = 2T$/TWa 
+    )
```

Comment:
Why not simply
```
q32_flexAdj(t,regi,te)$(teFlexTax(te))..
  vm_flexAdj(t,regi,te) 
  =e=
    (1 - v32_flexPriceShare(t,regi,te))
  * max(0, min(2, pm_SEprtive(t,regi,"seel")))$(    cm_flex_tax eq 1
                                                AND t.val ge 2025    )
;
```?

Reply:
I thought you could only use max/min-commands in data preparation, but not in equations when using NLP and not DNLP?

but if it works, your formulation definitely is shorter

Reply:
You can't use them with _variables_.  At the time of model generation, all the parameters are known to GAMS and collapsed together with all the numbers in the equation before being send off to the solver.